### PR TITLE
Feature/vote

### PR DESCRIPTION
**Pull request by JjuliaP**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=738)

## Code reviewers

- [ ] @dmytrorevak 
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [ ] @makarman
- [ ] @noobhunters
- [ ] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/vote/models.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
+"""test"""
```
Comment (lhalam) [inline eventually/vote/models.py]: update docstring for module

File: eventually/vote/models.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+from django.db import models
+
+
+class Vote(models.Model):
+    '''
```
Comment (lhalam) [inline eventually/vote/models.py]: for docstring use `"""` and not `'''`

File: eventually/vote/models.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+    MULTI_CHOICES = (
+        (u'O', u'One'),
+        (u'M', u'Multi'),
+    )
+
+    is_active = models.BooleanField(default=True)
+    is_extended = models.BooleanField(default=True)
+    title = models.CharField(max_length=100)
+    type = models.CharField(max_length=2, choices=MULTI_CHOICES)
```
Comment (lhalam) [inline eventually/vote/models.py]: do not use the name `type`

File: eventually/vote/models.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            'create_at': self.create_at,
+            'update_at': self.update_at
+        }
+
+    @staticmethod
+    def get_by_id(vote_id):
+        '''
+        Static method that return Vote object by id
+        :param id: id of element in model
+        :return: elements by id
```
Comment (lhalam) [inline eventually/vote/models.py]: what type?

File: eventually/vote/models.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        Static method that create new Vote object
+        :param myis_active: active vote or end
+        :param myis_extended:
+        :param mytitle: title of vote
+        :param mytype: type of vote - we can choose one answer or many
+        :return: new created field or None
+        '''
+
+
+        #Vote.objects.create()
```
Comment (lhalam) [inline eventually/vote/models.py]: remove unnecessary lines

File: eventually/vote/models.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+
+        try:
+            vote = Vote.objects.create(is_activate = myis_active,is_extended=myis_extended,title=mytitle,type=mytype)
+            #vote.create()
```
Comment (lhalam) [inline eventually/vote/models.py]: delete commented code.
in the repository, it should not be.

File: eventually/vote/models.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+            return vote
+        except:
+            return None
+
+
+
+
+
+    def update(self,vote_id):
```
Comment (lhalam) [inline eventually/vote/models.py]: cannot update id

**Review by lhalam — CHANGES_REQUESTED** [review]
forgot to add an to INSTALLED_APPS

**Comment by lhalam** [comment]
and create migrations/0001_initial.py file

File: eventually/vote/models.py [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+        '''
+        try:
+            return Vote.objects.get(id=vote_id)
+        except:
+            return None
+
+
+
+    @staticmethod
+    def create(myis_active,myis_extended,mytitle,mytype):
```
Comment (makarman) [inline eventually/vote/models.py]: use 'my_is_...'  not 'myis_...'

**Review by makarman — COMMENTED** [review]
use 'my_is_...'  not 'myis_...'

File: eventually/vote/models.py [hunk 9]
```diff
@@ -0,0 +1,210 @@
+"""test"""
+from django.db import models
+
+
+class Vote(models.Model):
+    '''
+        Create Vote model
+
+        Args:
+            * is_active (BooleanField): active vote or end
```
Comment (redmyt) [inline eventually/vote/models.py]: add `:param` and `type` directives for describing of class and functions attributes and parameters.

File: eventually/vote/models.py [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+            * title (CharField): title of vote
+            * type (CharField): type of vote - we can choose one answer or many
+            * create_at (TimeField)
+            * update_at (TimeField)
+
+
+    '''
+
+    MULTI_CHOICES = (
+        (u'O', u'One'),
```
Comment (redmyt) [inline eventually/vote/models.py]: Is it necessary to use `u` symbol before strings in Python3?

File: eventually/vote/models.py [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+
+        Args:
+        * vote (ForeignKey): id of vote
+        * text (CharField): variant to choose in vote
+        * create_at (TimeField)
+        * update_at (TimeField)
+
+    '''
+    vote = models.ForeignKey(Vote)
+    text = models.CharField(max_length=100)
```
Comment (redmyt) [inline eventually/vote/models.py]: It is better to use TextField for arbitrary text length.

File: eventually/vote/models.py (hunk 8 above)
Comment (redmyt) [inline eventually/vote/models.py]: add spaces between arguments

File: eventually/vote/models.py [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+    def update(self,vote_id):
+        '''
+        Method that update existed Vote object
+
+        :param vote_id:
+        :return: updated field or None
+        '''
+
+        try:
+            vote=Vote.objects.get(id=vote_id)
```
Comment (redmyt) [inline eventually/vote/models.py]: You should rewrite objects properties before `save` method

File: .gitignore [hunk 13]
```diff
@@ -100,5 +100,9 @@ ENV/
 # mypy
 .mypy_cache/
 
-# PyCharm
+#PyCharm
 .idea/
+
+#Sphinx
```
Comment (lhalam) [inline .gitignore]: update line 64, delete is it

File: eventually/vote/models.py [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+        Method that return Vote object in dictionary
+
+        :return:models-fields in dictionary
+        """
+        return {
+            'is_active': self.is_active,
+            'is_extended': self.is_extended,
+            'title': self.title,
+            'type': self.vote_type,
+            'create_at': self.create_at,
```
Comment (lhalam) [inline eventually/vote/models.py]: convert to timestamp

File: eventually/vote/models.py [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+        """
+        Static method that return Vote object by id
+
+        :param id(IntegerField): id of element in model
+        :return: object with element, searched by id
+        """
+        try:
+            return Vote.objects.get(id=vote_id)
+        except Vote.DoesNotExist:
+            return None
```
Comment (lhalam) [inline eventually/vote/models.py]: pass

File: eventually/vote/models.py [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+
+    @staticmethod
+    def create(my_is_active, my_is_extended, my_title, my_vote_type):
```
Comment (lhalam) [inline eventually/vote/models.py]: remove the prefix 'my_', add the default value if needed

File: eventually/vote/models.py [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+                    title=my_title,
+                    vote_type=my_vote_type)
+
+        try:
+            vote.save()
+            return vote
+        except ValueError:
+            return None
+
+    def update(self, my_is_active, my_is_extended, my_title, my_vote_type):
```
Comment (lhalam) [inline eventually/vote/models.py]: the prefix 'my_', add the default value if needed

File: eventually/vote/models.py [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+    def to_dict(self):
+        """
+        Method that return Answer object in dictionary
+
+        :return:models-fields in dictionary
+        """
+        return {
+            'vote': self.vote,
+            'text': self.text,
+            'create_at': self.create_at,
```
Comment (lhalam) [inline eventually/vote/models.py]: convert to timestamp

File: eventually/vote/models.py [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+            'update_at': self.update_at
+        }
+
+    @staticmethod
+    def get_by_id(answer_id):
+        """
+        Static method that return Answer object by id
+        :param id(IntegerField): id of element in model
```
Comment (lhalam) [inline eventually/vote/models.py]: add empty lines where needed

File: eventually/vote/models.py [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+        :return: object with element, searched by id
+        """
+        try:
+            return Answer.objects.get(id=answer_id)
+        except Answer.DoesNotExist:
+            return None
```
Comment (lhalam) [inline eventually/vote/models.py]: pass

File: eventually/vote/models.py [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+
+    @staticmethod
+    def create(my_vote, my_text):
```
Comment (lhalam) [inline eventually/vote/models.py]: the prefix 'my_', add the default value if needed

**Review by lhalam — CHANGES_REQUESTED** [review]

File: eventually/vote/models.py [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+    create_at = models.TimeField(auto_now_add=True, editable=False)
+    update_at = models.TimeField(auto_now=True)
+
+    def to_dict(self):
+        """
+        Method that return Vote object in dictionary
+
+        :return:models-fields in dictionary
+        """
+        return {
```
Comment (makarman) [inline eventually/vote/models.py]: add 'id' : self.id,

File: eventually/vote/models.py [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+    is_extended = models.BooleanField(default=True)
+    title = models.TextField(max_length=100)
+    vote_type = models.CharField(max_length=2, choices=MULTI_CHOICES)
+    create_at = models.TimeField(auto_now_add=True, editable=False)
+    update_at = models.TimeField(auto_now=True)
+
+    def to_dict(self):
+        """
+        Method that return Vote object in dictionary
+
```
Comment (makarman) [inline eventually/vote/models.py:53]: add example

File: .gitignore (hunk 13 above)
Comment (JjuliaP) [reply .gitignore]: done

File: eventually/vote/models.py (hunk 23 above)
Comment (JjuliaP) [reply eventually/vote/models.py:53]: done

File: eventually/vote/models.py (hunk 22 above)
Comment (JjuliaP) [reply eventually/vote/models.py]: done

File: eventually/vote/models.py (hunk 14 above)
Comment (JjuliaP) [reply eventually/vote/models.py]: done

File: eventually/vote/models.py (hunk 15 above)
Comment (JjuliaP) [reply eventually/vote/models.py]: done

File: eventually/vote/models.py (hunk 16 above)
Comment (JjuliaP) [reply eventually/vote/models.py]: done

File: eventually/vote/models.py (hunk 17 above)
Comment (JjuliaP) [reply eventually/vote/models.py]: done

**Review by lhalam — APPROVED** [review]

**Review by ssOleg — APPROVED** [review]

**Pull request merged**